File: wsnet2-unity/Assets/WSNet2/Scripts/Core/MessagePack/LobbyParams.cs
using System.Collections.Generic;
using MessagePack;

namespace WSNet2
{
    [MessagePackObject]
    public class CreateParam
    {
        [Key("room")]
        public RoomOption roomOption;

        [Key("client")]
        public ClientInfo clientInfo;

        [Key("emk")]
        public string encryptedMACKey;
    }

    [MessagePackObject]
    public class JoinParam
    {
        [Key("query")]
        public List<List<Query.Condition>> queries;

        [Key("client")]
        public ClientInfo clientInfo;

        [Key("emk")]
        public string encryptedMACKey;
    }

    [MessagePackObject]
    public class SearchParam
    {
        [Key("group")]
        public uint group;

        [Key("query")]
        public List<List<Query.Condition>> queries;

        [Key("limit")]
        public int limit;

        [Key("joinable")]
        public bool checkJoinable;

        [Key("watchable")]
        public bool checkWatchable;
    }

    [MessagePackObject]
    public class SearchByIdsParam
    {
        [Key("ids")]
        public string[] ids;

        [Key("query")]
        public List<List<Query.Condition>> queries;
    }
}

File: server/lobby/service/lobby_params.go
package service

import (
	"wsnet2/lobby"
	"wsnet2/pb"
)

type CreateParam struct {
	RoomOption pb.RoomOption `json:"room"`
	ClientInfo pb.ClientInfo `json:"client"`
	EncMACKey  string        `json:"emk"`
}

type JoinParam struct {
	Queries    []lobby.PropQueries `json:"query"`
	ClientInfo pb.ClientInfo       `json:"client"`
	EncMACKey  string              `json:"emk"`
}

type SearchParam struct {
	SearchGroup    uint32              `json:"group"`
	Queries        []lobby.PropQueries `json:"query"`
	Limit          uint32              `json:"limit"`
	CheckJoinable  bool                `json:"joinable,omitempty"`
	CheckWatchable bool                `json:"watchable,omitempty"`
}

type SearchByIdsParam struct {
	RoomIDs []string            `json:"ids"`
	Queries []lobby.PropQueries `json:"query"`
}

type SearchByNumbersParam struct {
	RoomNumbers []int32             `json:"numbers"`
	Queries     []lobby.PropQueries `json:"query"`
}
